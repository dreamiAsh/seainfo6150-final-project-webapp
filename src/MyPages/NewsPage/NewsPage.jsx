import React from "react";
import styles from "./NewsPage.module.css";

const NewsPage = (props) => {
  return (
    <div className={styles.box}>
            <p className={styles.title}>{props.title}</p>
            <div className={styles.dateBox}>
                <p className={styles.date}>{props.date}</p>
            </div>
            <div className={styles.imgBox}>
                <img className={styles.img} src={`https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%${props.imageId}.jpg&w=272&h=272&c=sc&poi=face&q=85`} alt=""/>
                <div className={styles.rightBox}>
                    <p className={styles.content}>{props.content}</p>
                </div>
            </div>
        </div>
  );
};

export default NewsPage;
