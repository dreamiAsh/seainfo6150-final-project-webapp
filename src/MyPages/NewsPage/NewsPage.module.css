.title{
    font-size: 28px;
    font-weight: bold;
    margin: 35px 60px 18px;
}
.dateBox{
    margin-left: 60px;
    margin-bottom: 40px;
    width: 90%;
    height: 25px;
}
.date{
    float: left;
    margin-right: 10%;
    color: gray;
}
.imgBox{
    margin-left: 60px;
    width: 94%;
    display: flexbox;
    flex-wrap: wrap; 
}
.img{
    width: 40%;
    float: left;
    margin-bottom: 140px;
}
.rightBox{
    float: left;
    width: 40%;
    margin-left: 3%;
}
.content{
    margin-top: 6%;
    line-height: 30px;
}