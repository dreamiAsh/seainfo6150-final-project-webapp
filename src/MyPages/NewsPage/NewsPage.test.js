import React from "react";
import { render, fireEvent } from "@testing-library/react";
import NewsPage from "./NewsPage.jsx";

it("renders the NewsPage component correctly", () => {
  const { container } = render(<NewsPage />);
  expect(container).toMatchSnapshot();
});