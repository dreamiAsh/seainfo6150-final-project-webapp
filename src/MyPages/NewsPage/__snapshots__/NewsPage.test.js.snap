// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the NewsPage component correctly 1`] = `
<div>
  <div
    class="box"
  >
    <p
      class="title"
    />
    <div
      class="dateBox"
    >
      <p
        class="date"
      />
    </div>
    <div
      class="imgBox"
    >
      <img
        alt=""
        class="img"
        src="https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%undefined.jpg&w=272&h=272&c=sc&poi=face&q=85"
      />
      <div
        class="rightBox"
      >
        <p
          class="content"
        />
      </div>
    </div>
  </div>
</div>
`;
